### ISSUE TITLE
Deep clustering loss

### ISSUE THREAD

**Issue by prashant45** [speechLabBcCuny/onssen#13]
Hi,

I have a doubt regarding the implementation of `loss_dc`.

https://github.com/speechLabBcCuny/onssen/blob/0479d78caeb0fefdb1160329fc443e91bcf6f6e5/losses/loss_dc.py#L27-L36

Why is this calculations performed by flattening the batch_size dim?

**Comment by nateanl** [comment]
Hi, this step is just for calculating the weights for each speaker. As you can see the embedding is again reshaped to (batch_size,  -1, embedding_dim).

**Comment by prashant45** [comment]
Is it necessary to calculate the weights considering all the samples in the mini-batch?

**Comment by nateanl** [comment]
I see. The weights should be batch-wise. I will keep it open and fix it later. If you want to help fix it, would you like to implement the W_MR weights mentioned in [this paper](https://www.merl.com/publications/docs/TR2018-005.pdf)? 
![image](https://user-images.githubusercontent.com/8653221/78242484-a407ed00-7514-11ea-9816-9d46f27ff296.png)
I won't be able to update the code soon but I will keep you posted once the code is done, thanks!

**Comment by prashant45** [comment]
I am trying to implement the same in keras

**Comment by nateanl** [comment]
Hi @prashant45, I have added the W_{MR} weight to [the loss function](https://github.com/speechLabBcCuny/onssen/blob/master/losses/loss_dc.py#L32). Thanks!
